pallets/settlement/src/lib.rs

```diff
@@ -1154,19 +1384,26 @@ impl<T: Trait> Module<T> {
                             SettlementDID.as_id(),
                             instruction_id,
                         ));
+                        // TODO: do we want to emit the error reason as well?
+                        // match reason {
+                        //     FailureReason::BaseTransfer => Err(DispatchError::from(Error::<T>::TODONewError)),
+                        //     FailureReason::ConfidentialTransfer => Err(DispatchError::from(Error::<T>::InvalidMercatTransferProof)),
+                        //     FailureReason::InsufficientMercatAuthorizations => Err(DispatchError::from(Error::<T>::TODONewError)),
+                        //     FailureReason::UndefinedLeg => Err(DispatchError::from(Error::<T>::UndefinedLeg)),
+                        // }
```

Comment:
Do we?

Reply:
CC @vkomenda @satyamakgec ?

Reply:
Yes, we can do for better application development. But we can add a new task to handle this. I don't want to stretch this PR anymore.